### PR TITLE
Allow parsing of offxml strings

### PR DESCRIPTION
**Pull request by jthorton**
This PR allows an offxml string to be used as a small molecule force field for the `SystemGenerator` and `SMIRNOFFTemplateGenerator` see the example which now works with this PR.

```python
"Try and load a OFFXML string into a system generator object"
from openmmforcefields.generators import SystemGenerator
from openff.toolkit.typing.engines.smirnoff import ForceField
from openff.toolkit.topology import Molecule
import openmm

# load the ff
ff = ForceField("openff-2.0.0.offxml")
# create a system generator
system_gen = SystemGenerator(
    forcefields=['amber/ff14SB.xml', 'amber/tip3p_standard.xml'],
    small_molecule_forcefield=ff.to_string()
)
mol = Molecule.from_smiles("CC")
system = system_gen.create_system(topology=mol.to_topology().to_openmm(), molecules=mol)
with open("system.xml", "w") as output:
    output.write(openmm.XmlSerializer.serialize(system))
```

### CODE REVIEW

**Comment by jthorton** [comment]
Confirmed this works when running the solvation leg of a benchmark free energy calculation via openfe.

**Comment by mikemhenry** [comment]
@jthorton I did a merge commit to bring in a fix that was needed, let me know if it looks okay!

**Comment by jthorton** [comment]
Thanks @mikemhenry looks like it changed the exception message slightly which is fine for me. Just want to check if my use of the walrus `:=` was okay? I think this will only work with python 3.8+

**Comment by mikemhenry** [comment]
I am happy with using python 3.8+ language features. We can even use python 3.9+ language features since we follow NEP29 (or try to)

**Comment by mattwthompson** [comment]
OpenFF is pretty good about using it, so congrats - you've inherited it!

**Comment by mikemhenry** [comment]
Ah classic, can't run this from a fork. @jthorton I will push this to a branch in the repo so we can make sure it passes in CI

**Comment by mikemhenry** [comment]
@jthorton I don't know if this trick will keep working if you make a commit, so ping me if you do and CI stops running

**Comment by mikemhenry** [comment]
Looks like it passes :rocket: @ijpulidos would you mind reviewing this one? Should we wait for #293 first?

**Review by mikemhenry — APPROVED** [review]

**Comment by mikemhenry** [comment]
I don't think we need to wait on any other PRs, just need another review from @ijpulidos

File: openmmforcefields/generators/template_generators.py
Comment (ijpulidos) [inline openmmforcefields/generators/template_generators.py:1304]: I think we want to improve the error message here, since as far as I can see it's not now that we don't find the ff on the install paths but it's rather that the provided string is not correctly formatted as a ffxml string.

Is it too crazy to add some more logic here such that we can have specific error messages as to what is causing this? It doesn't seem too hard to accomplish, as far as I can tell.

**Review by ijpulidos — COMMENTED** [review]
This is looking great! I just added some comment about the error message when you get a poorly formatted string.

File: openmmforcefields/generators/template_generators.py
Comment (jthorton) [reply openmmforcefields/generators/template_generators.py]: I have updated the error to reflect that a missing file or poorly formatted string could cause this error. As for specific messages on the formatting of the forcefield I would prefer to leave this to the `openff.toolkit.ForceField` and just raise the error from there as it's quite descriptive itself.

**Pull request merged**
